app/src/src_nonfreenet/org/breezyweather/sources/cwa/CwaGeoLookup.kt

```diff
@@ -0,0 +1,217 @@
+}
+
+// Temperature normals are only available at 26 stations (out of 700+).
+// They are not available from the main weather API call,
+// and must be called with a different endpoint with the exact station ID.
+// This list allows matching a location to the nearest of those 26 stations.
+//
+// Information is updated once every 10 years. Last update was after 2020.
+//
+// Source (last checked 2024-05-29):
+// https://opendata.cwa.gov.tw/dataset/climate/C-B0027-001
+val CWA_NORMALS_STATIONS = mapOf<String, Map<String, Double>>(
+    "466880" to mapOf("lat" to 24.997646, "lon" to 121.44202),  // 板橋 BANQIAO
+    "466900" to mapOf("lat" to 25.164888, "lon" to 121.448906), // 淡水 TAMSUI
+    "466910" to mapOf("lat" to 25.182587, "lon" to 121.52973),  // 鞍部 ANBU
+    "466920" to mapOf("lat" to 25.037659, "lon" to 121.514854), // 臺北 TAIPEI
+    "466930" to mapOf("lat" to 25.162079, "lon" to 121.54455),  // 竹子湖 ZHUZIHU
+    "466940" to mapOf("lat" to 25.133314, "lon" to 121.74048),  // 基隆 KEELUNG
+    "466950" to mapOf("lat" to 25.627975, "lon" to 122.07974),  // 彭佳嶼 PENGJIAYU
+    "466990" to mapOf("lat" to 23.975128, "lon" to 121.61327),  // 花蓮 HUALIEN
+    "467060" to mapOf("lat" to 24.596737, "lon" to 121.85737),  // 蘇澳 SU-AO
+    "467080" to mapOf("lat" to 24.763975, "lon" to 121.75653),  // 宜蘭 YILAN
+    "467300" to mapOf("lat" to 23.25695, "lon" to 119.667465),  // 東吉島 DONGJIDAO
+    "467350" to mapOf("lat" to 23.565502, "lon" to 119.563095), // 澎湖 PENGHU
+    "467410" to mapOf("lat" to 22.993238, "lon" to 120.20477),  // 臺南 TAINAN
+    "467440" to mapOf("lat" to 22.565992, "lon" to 120.315735), // 高雄 KAOHSIUNG
+    "467480" to mapOf("lat" to 23.495926, "lon" to 120.43291),  // 嘉義 CHIAYI
+    "467490" to mapOf("lat" to 24.145737, "lon" to 120.684074), // 臺中 TAICHUNG
+    "467530" to mapOf("lat" to 23.508207, "lon" to 120.81324),  // 阿里山 ALISHAN
+    "467540" to mapOf("lat" to 22.355675, "lon" to 120.903786), // 大武 DAWU
+    "467550" to mapOf("lat" to 23.487614, "lon" to 120.95952),  // 玉山 YUSHAN
+    "467571" to mapOf("lat" to 24.827852, "lon" to 121.01422),  // 新竹 HSINCHU
+    "467590" to mapOf("lat" to 22.003897, "lon" to 120.74634),  // 恆春 HENGCHUN
+    "467610" to mapOf("lat" to 23.097486, "lon" to 121.37343),  // 成功 CHENGGONG
+    "467620" to mapOf("lat" to 22.036968, "lon" to 121.55834),  // 蘭嶼 LANYU
+    "467650" to mapOf("lat" to 23.881325, "lon" to 120.90805),  // 日月潭 SUN MOON LAKE
+    "467660" to mapOf("lat" to 22.75221, "lon" to 121.15459),   // 臺東 TAITUNG
+    "467770" to mapOf("lat" to 24.256002, "lon" to 120.523384)  // 梧棲 WUQI
+)
+
+// UV Index observations are only available at 29 stations (out of 700+).
+// They are not available from the main weather API call,
+// and must be called with a different endpoint with the exact station ID.
+// This list allows matching a location to the nearest of those 29 stations.
+//
+// Source (last checked 2024-05-29):
+// https://opendata.cwa.gov.tw/dataset/observation/O-A0003-001
```

Comment:
This link shows that this goes further than just UVI, here is what is listed:
AirPressure、AirTemperature、GustInfo、WindDirection、PeakGustSpeed、WindSpeed、Precipitation、RelativeHumidity、SunshineDuration、UVI

Could you clarify why UV needs a special case? If some stations don't have all kind of data, we should not mix stations between each other as they can be opposite, and leave the missing information empty instead.

Reply:
Unfortunately the CWA only record UV measurements at fewer than 30 stations out of the 700+. In theory the measurements could be valid over longer distances, as variations of solar radiation over longer distances should be less than those of other weather elements.

But I can see why you wouldn't feel comfortable including UV Index from nearby stations. I'm fine removing it. I -seem to remember Breezy would "estimate" the daytime UV Index from the forecast data – not ideal, but there are no ideal solutions here, only preferable ones. And I defer to your preference. 😉

BTW, the above is an older endpoint that does not take geographic coordinates, only station IDs, but it can output real-time UV observations. I did not use this endpoint for the main request because it would require downloading and looping through an entire 700+ station list to identify the nearest one, which you indicated in other threads that you do not want happening in the app.

The newer endpoint (used for the main request) takes geographic coordinates, but does not output UV Index measurements even if the station has it.

Reply:
> The newer endpoint (used for the main request) takes geographic coordinates, but does not output UV Index measurements even if the station has it.

I believe you should contact them to let them know that this information is missing, they probably already have the data and forgot to add it. That would be the easiest path for everyone